### PR TITLE
Add support for setting the channel input filter

### PR DESCRIPTION
**Pull request by dberlin**
## Pull Request template

Please, Make sure that your PR is not a duplicate.
Search among the [Pull request](https://github.com/stm32duino/Arduino_Core_STM32/pulls) before creating one.

IMPORTANT: Please review the [CONTRIBUTING.md](../CONTRIBUTING.md) file for detailed contributing guidelines.

Thanks for submitting a pull request.
Please provide enough information so that others can review your pull request:

**Summary**

This PR implements the following **feature**

- Support for input channel filtering in HardwareTimer

Currently, HardwareTimer sets the input channel filter to none, any time the mode of the timer is set.
This PR enables the input filter to be set along with the mode.

It is possible to enable setting the input filter separately from the mode if wanted.

**Validation**

Tested with a waveform generator vs sketch outputting pulse widths + levels

### CODE REVIEW

File: cores/arduino/HardwareTimer.h [hunk 1]
```diff
@@ -90,6 +90,25 @@ typedef enum {
   PERCENT_COMPARE_FORMAT, // used for Dutycycle
 } TimerCompareFormat_t;
 
+typedef enum {
+  FILTER_NONE = 1,  // No filter
```
Comment (ABOSTM) [inline cores/arduino/HardwareTimer.h:94]: Hi @dberlin 
Thanks for this pull request. It seems functionally good to me, but I wonder whether we can simplify it:
Can we get ride of function getICFBitsFromFilterMode() by setting `FILTER_NONE = 0`  ?
And thus enum `ChannelInputFilter_t `directly matches with ICF bit

**Comment by fpistm** [comment]
Hi @dberlin
thanks for this PR.
Could you also share your test sketch, please ?

File: cores/arduino/HardwareTimer.h (hunk 1 above)
Comment (dberlin) [reply cores/arduino/HardwareTimer.h]: Done!

**Comment by dberlin** [comment]
> Hi @dberlin thanks for this PR. Could you also share your test sketch, please ?

Sure.

Removing a whole bunch, it's attached:
Note - i use platformio, and define the c++ standard flag to gnu++17.
The sketch also uses the streaming library, but this could be easily changed.

I parse the pulses out and verify the results.
[main.cpp.txt](https://github.com/stm32duino/Arduino_Core_STM32/files/12721383/main.cpp.txt)

**Review by ABOSTM — APPROVED** [review]
Thanks @dberlin 
LGTM

**Pull request merged**